### PR TITLE
Allow for Okta Assignments with empty targets

### PR DESCRIPTION
**Pull request by smallinsky**
### What

Allow for empty assignments. 

For the product perspective assignment with empty  targets should be never created: 
- [AccessRequestFlow](https://github.com/gravitational/teleport.e/blob/f014812393a101d7a0070e2018e22861d98c77bb/lib/okta/accessrequests.go#L576): 
- okta login [hook](https://github.com/gravitational/teleport.e/blob/master/lib/okta/users.go#L198): 

and empty assignment does nop operation during assignment processing   [cleanupTargets](https://github.com/gravitational/teleport.e/blob/f0d670881e1a1f472f7f4a6c1e2d5477f6e5ffa3/lib/okta/assignmentprocessor.go#L411) [processTargets](https://github.com/gravitational/teleport.e/blob/f0d670881e1a1f472f7f4a6c1e2d5477f6e5ffa3/lib/okta/assignmentprocessor.go#L359)

The idea it to leverage empty target assignment in order to reduce numbers of OKTA API calls during handling assigment cleanup after DELTA  `OKta app,groups premissons` delta recalculation.  

Related: https://github.com/gravitational/teleport.e/pull/3960

### CODE REVIEW

**Review by r0mant — APPROVED** [review]

File: api/types/okta.go [hunk 1]
```diff
@@ -385,10 +385,6 @@ func (o *OktaAssignmentV1) CheckAndSetDefaults() error {
 		return trace.BadParameter("user must not be empty")
 	}
 
-	if len(o.Spec.Targets) == 0 {
```
Comment (codingllama) [inline api/types/okta.go:388]: This is OK to remove because all places where targets are relevant already do their own checks? Is that it?

**Review by codingllama — APPROVED** [review]
Approved but I can't say I have a lot of context here. It would be best if @mdwn had a look.

**Comment by mdwn** [comment]
I think the necessity of this will be dependent on whether or not the referencing counting logic in the Okta service accomplishes the same goal here. @smallinsky is going to evaluate.

**Review by tcsc — APPROVED** [review]

File: api/types/okta.go (hunk 1 above)
Comment (smallinsky) [reply api/types/okta.go:388]: Yes,  empty targets should be evaluated as void operation  during assignment processing flow: 
 [cleanupTargets](https://github.com/gravitational/teleport.e/blob/f0d670881e1a1f472f7f4a6c1e2d5477f6e5ffa3/lib/okta/assignmentprocessor.go#L411) [processTargets](https://github.com/gravitational/teleport.e/blob/f0d670881e1a1f472f7f4a6c1e2d5477f6e5ffa3/lib/okta/assignmentprocessor.go#L359)

**Review by mdwn — APPROVED** [review]

**Pull request merged**